tiledb/sm/c_api/tiledb.cc

```diff
@@ -3640,13 +3640,17 @@ int32_t tiledb_subarray_alloc(
         ctx->ctx_->storage_manager());
     (*subarray)->is_allocated_ = true;
   } catch (...) {
+    auto st = Status_Error("Failed to construct TileDB subarray object");
+    LOG_STATUS(st);
+    save_error(ctx, st);
+    return TILEDB_ERR;
```

Comment:
There's a memory leak possible here. When the allocation for `*subarray` succeeds and that for `(*subarray)->subarray_` fails, it's necessary to deleted `*subarray`.

#3222 has an example of this.